typescript/cli/src/rebalancer/monitor/Monitor.ts

```diff
@@ -1,19 +1,37 @@
 import EventEmitter from 'events';
 
 import { IRegistry } from '@hyperlane-xyz/registry';
-import { MultiProtocolProvider } from '@hyperlane-xyz/sdk';
-import { WarpCore } from '@hyperlane-xyz/sdk';
-import { objMap, objMerge } from '@hyperlane-xyz/utils';
+import { MultiProtocolProvider, WarpCore } from '@hyperlane-xyz/sdk';
+import { objMap, objMerge, sleep } from '@hyperlane-xyz/utils';
 
 import { IMonitor, MonitorEvent } from '../interfaces/IMonitor.js';
 
+class WrappedError extends Error {
+  constructor(message: string, originalError?: Error) {
+    super(message);
+    this.name = 'WrappedError';
+
+    // Preserve the stack trace of the original error if available
+    if (originalError?.stack) {
+      this.stack = `${this.stack}\nCaused by: ${originalError.stack}`;
+    }
+  }
+}
+
+export class MonitorStartError extends WrappedError {
```

Comment:
This could be moved to its own errors.ts file

Reply:
it's so Monitor specific that I don't find the need to do it